Remove execute mode bit from source files

cbuescher opened the pull request:
Some source files seem to have the execute bit (a+x) set, which doesn't
really seem to hurt but is a bit odd. This change removes those, making
the permissions similar to other source files in the repository.

Commit:
Remove execute mode bit from source files

Some source files seem to have the execute bit (a+x) set, which doesn't
really seem to hurt but is a bit odd. This change removes those, making
the permissions similar to other source files in the repository.

jpountz reviewed (approved):
Change looks good, but I wonder if we should do something to prevent such files from being checked in in the first place.

cbuescher commented:
No idea if this should be part of the `precommit` checks. I found these by accident playing around with `find` and trying various `perm` settings: `find . -perm -a+x -name "*.java` 
Not sure how anything like that would translate to an automated check and if it is really worth it. Maybe @rjernst has an opinion? I think I'm going to merge this, regardless.

rjernst commented:
Adding a check should be simple, I think. Just a new task created in PrecommitTasks, which operates on the source files (see gradle's SourceSet)

Commit:
Add file permissions checks to precommit task

This adds a check for source files that have the execute bit set to the
precommit task.

cbuescher commented:
@jpountz I found a few more files that have the execute bit set (some yml rest tests and some hunspell resources), also added a task to the gradle precommit task to check this in the future and error when there are source files that are exectuable (with the exception of *.sh files).

Maybe @rjernst can take a look at that task? I mostly used the ForbiddenPatternsTask as a template and adapted it a bit. Not sure if I't doing it right, first time dabbling with groovy/gradle task but it seems to work for this simple task.
I can also separate the two commits if the file permission changes are alright but the task needs some more iterations.

rjernst commented:
This looks ok, but I think it will fail on windows (java will throw an exception that you can't get a posix view)? How long does this check take? I wonder if this is something we should run periodically (eg once a week) rather than in precommit since it doesn't actually hurt, but it is yet another thing that will require processing time for precommit (which has continually gotten slower, even though the idea is to do a quick check before committing).

cbuescher commented:
> I think it will fail on windows

I can try that in a VM, I got some code from the PluginInstallCommand, thought that would work on windows. Will see and report back

> I wonder if this is something we should run periodically (eg once a week) rather than in precommit

I agree, thats why I initially didn't add a check, also because it doesn't really hurt but is a nice-to-have. How your adding weekly checks work? Do we have any infra for that?

rjernst commented:
> Do we have any infra for that?

There is a top level task we run once a day for "branch consistency". But we could add a separate task for this maybe, or rename that task (that is more meant for things to make sure we don't forget stuff, eg version bumps without adding versions in code).

cbuescher commented:
> I wonder if this is something we should run periodically (eg once a week) rather than in precommit

An argument of running in precommit would be that we might want to catch those accidental changes in permissions in PR testing for example. I don't think the task itself takes a lot of time, is there an easy way of measuring this? Don't know how long setup of these tasks take though.

Commit:
Exclude check from running on windows

rjernst commented on buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/FilePermissionsTask.groovy:
```diff
@@ -0,0 +1,88 @@
+
+    @TaskAction
+    void checkInvalidPermissions() {
+        if (Os.isFamily(Os.FAMILY_WINDOWS) == false) {
```
Instead, in the constructor, use `onlyIf { Os.isFamily(Os.FAMILY_WINDOWS) }`. Then the task will show up as `SKIPPED` instead of looking like it ran (when it was in fact skipped internally).

rjernst reviewed (approved):
Looks good, one last request.

Commit:
Move os condition to ctor

cbuescher commented:
Closed by 318b0af and a9392f6
